artmanflow/steps/java_sources_staging_guest.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+import os
+
+import github3
+
+try:
+    from common import GuestStepProperties, ConfigUtils, GitUtils, BaseGuest
+except ImportError:
+    from artmanflow.steps.common import \
+        GuestStepProperties, ConfigUtils, GitUtils, BaseGuest
+
+
+class JavaSourcesStagingGuest(BaseGuest):
+    def execute(self):
+        try:
+            self.before_execute()
+            staging_name = self.checkout_git_output_repo(
+                self._config['staging'])
+            client_folders = self._extract_client_folders()
+            self._copy_artifacts_to_staging(staging_name, client_folders)
+            self._fix_staging_versions(staging_name)
+            self._build_and_test(staging_name)
+            self._git_commit_and_push(staging_name)
+            pr_url = self._post_pr()
+            self._dump_output({'pr_url': pr_url})
+            self.after_execute(self._config['debug_mode'])
+        except Exception as e:
```

Comment:
Please catch the specific exception.

Reply:
There is no specific exception. The point of this catch block is to output error_log into web console and exit. It is just to show in the console that execution failed (in command line, when command fails you can see it failed and command prompt is active again, in web it is hard to distinguish between "currently running" app and a failed app).